scripts/release-merge-to-main.ts

```diff
@@ -0,0 +1,178 @@
+//   GITHUB_TOKEN: GitHub authorization token.
+//     Must have write permissions for `actions`, `contents`, `pull_requests`.
+
+import fs from 'fs';
+
+import { RequestError } from 'octokit';
+
+import { getOctokit } from './lib/dependencies';
+
+/**
+ * Valid value for an environment variable.
+ */
+type EnvironmentVariableName =
+  'GITHUB_REPOSITORY' |
+  'GITHUB_EVENT_PATH' |
+  'GITHUB_TOKEN';
+
+/**
+ * Partial contents of the event payload, for a release event.
+ */
+interface GitHubReleasePayload {
+  release: {
+    tag_name: string;
+  }
+}
+
+/**
+ * The name of the branch to merge into.
+ */
+const base = 'main';
+
+/**
+ * Read the environment variable, or throw an error.
+ * @param variable The environment variable to look up.
+ * @returns The environment variable value.
+ */
+function getEnv(variable: EnvironmentVariableName): string {
+  const result = process.env[variable];
+
+  if (typeof result !== 'string') {
+    throw new ReferenceError(`Environment variable ${ variable } is not set`);
+  }
+
+  return result;
+}
+
+/**
+ * Ensure that the given branch exists, and points to the given tag.
+ * @param owner The repository owner.
+ * @param repo The repository name (without the owner).
+ * @param branchName The name of the branch.
+ * @param tagName The name of the tag.
+ */
+async function ensureBranch(owner: string, repo: string, branchName: string, tagName: string): Promise<void> {
+  const ref = `heads/${ branchName }`;
+  const { git } = getOctokit().rest;
+  const { data: tagRef } = await git.getRef({
+    owner, repo, ref: `tags/${ tagName }`,
+  });
+  const { sha } = tagRef.object;
+
+  try {
+    const { data: existingBranch } = await git.getRef({
+      owner, repo, ref,
+    });
+
+    if (existingBranch.object.sha !== sha) {
+    // Branch exists, but points at the wrong hash; update it.
+      await git.updateRef({
+        owner, repo, ref, sha,
+      });
+    }
+  } catch (ex) {
+    if (!(ex instanceof RequestError) || ex.status !== 404) {
+      throw ex;
+    }
+    // Branch does not exist; create it.
+    await git.createRef({
+      // Only this API takes a `refs/` prefix; get & update omit it.
+      owner, repo, ref: `refs/${ ref }`, sha,
+    });
+  }
+}
+
+/**
+ * Locate an existing pull request.
+ * @param owner The repository owner.
+ * @param repo The repository name (without the owner).
+ * @param branch The branch to merge from (i.e. the release branch).
+ * @returns The found pull request, or undefined.
+ */
+async function findExisting(owner: string, repo: string, branch: string) {
+  const fullRepo = `${ owner }/${ repo }`;
+  const query = `type:pr is:open repo:${ fullRepo } base:${ base } head:${ branch } sort:updated`;
+  const result = await getOctokit().rest.search.issuesAndPullRequests({ q: query });
+
+  for (const item of result.data.items) {
+    // Must be an open item, and that item must be a pull request.
+    if (item.state !== 'open' || !item.pull_request) {
```

Comment:
I agree with keeping this condition in here for defensive purposes, but since the query explicitly selects for open PRs, does GitHub return any other issues from this request?

Reply:
No, I don't expect this to actually match anything.